--- ./SR.py.orig	2016-05-10 14:45:56.000000000 +0800
+++ ./SR.py.new	2016-05-10 14:49:21.000000000 +0800
@@ -469,7 +469,10 @@
             self.mpathmodule.deactivate()
 
     def _pathrefresh(self, obj):
-        self.dconf['device'] = self.mpathmodule.path(self.SCSIid)
+        if self.dconf.has_key('port') and self.dconf['port'] == '6789':
+            self.dconf['device'] = self.mpathmodule.path_nomp(self.SCSIid)
+        else:
+            self.dconf['device'] = self.mpathmodule.path(self.SCSIid)
         super(obj, self).load(self.uuid)
 
     def _setMultipathableFlag(self, SCSIid=''):
